lib/zipline/zip_generator.rb

```diff
@@ -44,9 +44,7 @@ def normalize(file)
         {file: File.open(file.path)}
       elsif is_io?(file)
         {file: file}
-      elsif defined?(ActiveStorage::Blob) && file.is_a?(ActiveStorage::Blob)
-        {url: file.service_url}
-      elsif defined?(ActiveStorage::Attachment) && file.is_a?(ActiveStorage::Attachment)
```

Comment:
why are you removing ActiveStorage::Attachment? shouldn't you just be changing it to ActiveStorage::Attachment::One ?

Reply:
Actually I've found `Attached` may be the right class. We want to support both `One` and `Many`

https://api.rubyonrails.org/classes/ActiveStorage/Attached/One.html
https://api.rubyonrails.org/classes/ActiveStorage/Attached/Many.html

Reply:
What would the behavior here be with many? I feel like they'd need to hand us a specific blob. If they handed us a collection of files which one would go in the filename?

Reply:
So I've looked at the source and I believe Attachment::Many would be a proxy to foo_attachments, which is going to be an array of ::Attached

so actually I think this won't work.

I believe the statement we want here is actually:

`(defined?(ActiveStorage::Attachment) && file.is_a?(ActiveStorage::Attachment) ||
(defined?(ActiveStorage::Attached::One) && file.is_a?(ActiveStorage::Attached::One)`

this would work with `User.avatar` or `User.avatars[0]`